### PR TITLE
the yung boy is fullstack

### PR DESCRIPTION
**Pull request by benauriemma**

### CODE REVIEW

File: rackcity/api/objects/rack_range.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
 from rest_framework import serializers
 from rackcity.models.rack import validate_row_letter
+from rackcity.models import Datacenter
 
 
 class RackRangeSerializer(serializers.Serializer):
+    datacenter = serializers.PrimaryKeyRelatedField(
+        label='datacenter',
+        queryset=Datacenter.objects.all()
+    )
```
Comment (benauriemma) [inline rackcity/api/objects/rack_range.py]: This is from my other PR, I pulled it into this branch

Comment (benauriemma) [reply rackcity/api/objects/rack_range.py]: All the python stuff is

File: src/utils/utils.tsx [hunk 2]
```diff
@@ -42,6 +43,12 @@ export interface RackResponseObject {
   rack: RackObject;
   instances: Array<InstanceObject>;
 }
+
+export interface DatacenterObject extends ElementObject {
+  name: string;
+  abbreviation: string;
+}
+
```
Comment (clairecfu) [inline src/utils/utils.tsx]: wooo good interface

File: src/forms/formUtils.tsx [hunk 3]
```diff
@@ -48,6 +48,22 @@ export const filterRack: ItemPredicate<RackObject> = (
...
   }
 };
+export const filterDatacenter: ItemPredicate<DatacenterObject> = (
+  query,
+  datacenter,
+  _index,
+  exactMatch
+) => {
+  const name = datacenter.name.toLowerCase();
+  const abbreviation = datacenter.abbreviation;
```
Comment (clairecfu) [inline src/forms/formUtils.tsx]: should have `toLowerCase()` on `datacenter.abbreviation` as well

**Pull request merged**